nexus/db-queries/src/db/datastore/auth.rs

```diff
@@ -0,0 +1,81 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Implements the [Storage] interface for [nexus_auth] integration.
+
+use crate::db;
+use crate::db::error::public_error_from_diesel;
+use crate::db::error::ErrorHandler;
+
+use async_bb8_diesel::AsyncRunQueryDsl;
+use diesel::prelude::*;
+use nexus_auth::context::OpContext;
+use nexus_auth::storage::Storage;
+use nexus_db_model::IdentityType;
+use nexus_db_model::RoleAssignment;
+use omicron_common::api::external::Error;
+use omicron_common::api::external::ResourceType;
+use uuid::Uuid;
+
+#[async_trait::async_trait]
+impl Storage for super::DataStore {
+    /// Return the built-in roles that the given built-in user has for the given
+    /// resource
+    async fn role_asgn_list_for(
```

Comment:
This method was moved exactly from `nexus/db-queries/src/db/datastore/role.rs`

Reply:
nit: could spell out `assign`, but I also understand not wanting to introduce that to the diff if this is used in a lot of places.

Reply:
Grepping for `role_asgn`, this appears to be a pattern used beyond this method. I agree with you that I'd rather it be fully spelled out, but I'm gonna punt on this to avoid touching several files that aren't currently part of this PR?

Reply:
That works for me